front/lib/api/poke/plugins/workspaces/buy_programmatic_usage_credits.ts

```diff
@@ -29,29 +33,48 @@ const BuyCreditPurchaseArgsSchema = z.object({
 export const buyProgrammaticUsageCreditsPlugin = createPlugin({
   manifest: {
     id: "buy-programmatic-usage-credits",
-    name: "Buy Programmatic Usage Credits",
+    name: "Buy Committed Credits",
     description:
-      "Purchase programmatic usage credits for enterprise customers. The purchase will be added to the customer's subscription and paid on the next billing cycle.",
+      "Purchase committed credits for enterprise customers. Committed credits are consumed after free credits and before pay-as-you-go (PAYG) credits. The purchase will be added to the customer's subscription and paid on the next billing cycle.",
     resourceTypes: ["workspaces"],
     args: {
       amountDollars: {
         type: "number",
         label: "Credit Amount ($)",
         description:
-          "Programmatic usage credits amount, in usd, not billed amount. So without VAT, currency conversion and discounts",
+          "Committed credits amount in USD (not billed amount). Excludes VAT, currency conversion and discounts",
+      },
+      overrideDiscount: {
+        type: "boolean",
+        variant: "toggle",
+        label: "Override Default Discount",
+        async: true,
+        asyncDescription: true,
       },
       discountPercent: {
         type: "number",
         async: true,
         label: "Billing Discount (%)",
-        description:
-          "Discount percentage to apply (0-100) on the billed amount",
+        description: "Discount applied to the actual credit purchase",
+        dependsOn: { field: "overrideDiscount", value: true },
+      },
+      startDate: {
+        type: "date",
+        async: true,
+        label: "Start Date",
+        description: "When the credits become active. Default: today",
+      },
+      expirationDate: {
+        type: "date",
+        async: true,
+        label: "Expiration Date",
+        description: "When the credits expire. Default: 1 year from start date",
       },
       confirm: {
         type: "boolean",
         label: "Confirm Purchase",
         description:
-          "I understand that running this plugin will add a purchase in the customer's subscription, which will be paid on next billing cycle",
+          "I understand that running this plugin will add committed credits to the customer's subscription, which will be paid on next billing cycle (next month 99% of the time).",
```

Comment:
nit (in general, I favor short and 95% correct to longer and 99% correct. And avoid people wondering)
```suggestion
          "I understand that running this plugin will add committed credits to the customer's subscription, which will be paid on next billing cycle.",
```